src/v/bytes/details/io_iterator_consumer.h

```diff
@@ -113,15 +112,14 @@ class io_iterator_consumer {
 
     template<typename Consumer>
     // clang-format off
-    CONCEPT(requires requires(Consumer c, const char* src, size_t max) {
+    requires requires(Consumer c, const char* src, size_t max) {
```

Comment:
I've looked at the first dozen or so of these and I think that we should also drop the `// clang-format off/on` markup.

Reply:
wanted to minimize the code churn -- it seems that clang-12 does not like the `requires` statement, probably we will have to deal with it anyhow. let's remove the `clang-format off` guards as well.

Reply:
i formated the tree with clang-format-14, but clang-format-12 is still not satisfied with the outcome.

Reply:
Forget clang-format-12, `dev` branch now officially requires the library parts of concepts, so clang-13 is a likely a minimum, but clang-14 is what should be expected.

Reply:
@tchaikov if you end up removing the clang-format on/off please submit as two commits: one with the raw changes and one clang-format-only commit (unless there isn't really much change, of course).